Classes/Indexer/NodeIndexer.php

```diff
@@ -274,6 +274,13 @@ public function indexNode(NodeInterface $node, $targetWorkspaceName = null): voi
         $handleNode = function (NodeInterface $node, Context $context) use ($targetWorkspaceName, $indexer) {
             $nodeFromContext = $context->getNodeByIdentifier($node->getIdentifier());
             if ($nodeFromContext instanceof NodeInterface) {
+                if ($node->getPath() !== $nodeFromContext->getPath()) {
+                    // If the node from context does have a different path, purge the context cache and re-fetch
+
+                    // TODO: find a better way to handle this
```

Comment:
What is wrong with this solution? Would you like to avoid flushing the full cache?

Reply:
Indeed; I will prefer flushing the cache just for this specific node, but I was not able to figure out a good way to handle that. Do you have an idea here?

And maybe it should be safe(r) to just flush the cache **always**, regardless of the "node path was changed" condition. What do you think?

### TL;DR

I can remember that back then, in 2016 (!) as I have implemented this initially, it took me some time of hair pulling to track down this issue.. I still do not quite understand why the cached data does not cause any issues when the node path was not being changed; maybe the neos context node caching behavior has some flaws, using the node path for invalidation or stuff like that?